### ISSUE TITLE
⌨ Focus after toggling role on nav bar

### ISSUE THREAD

**Issue by vd1992** [GCTC-NTGC/gc-digital-talent#11994]
## ⌨ Accessibility Issue

Toggle your active role should keep focus there so you can access the changed links. Currently focus goes below the nav bar making you have to go back up

## 📋 Steps to Reproduce

1. Tab to role select
2. Switch role
3. Notice where focus ends up next, preventing quickly getting to a new link

## 📸 Screenshot

Start here

![image](https://github.com/user-attachments/assets/aad56dd6-e8d0-42e6-81d4-5955d471818a)

Switch to admin with the goal of let us say, accessing the users page

![image](https://github.com/user-attachments/assets/bad48efe-d16b-463d-9cd6-c71a94230a21)

Press tab and focus ends up on a button link below

![image](https://github.com/user-attachments/assets/78520ffb-959d-45e8-ba6a-b90f091f2af8)

## ✅ Acceptance Criteria

- [ ] Focus stays around nav bar

**Comment by esizer** [comment]
This is going to be very tricky :sweat_smile: 

We auto focus the `h1` on a page when changing routes to announce to screen readers about the new context since it is missing due to the fact that we have an SPA. 

I also found [this interaction very confusing](https://github.com/GCTC-NTGC/gc-digital-talent/issues/11978) while doing screen reader testing and I honestly don't know what to do :woman_shrugging:

**Comment by tristan-orourke** [comment]
We will prioritize #11978 over this, since its not clear if this is an issue with improved headers or how to best improve it.
